tensorpipe/test/channel/channel_test.h

```diff
@@ -202,6 +203,62 @@ class ChannelTestCase {
 
 template <typename TBuffer>
 class ClientServerChannelTestCase : public ChannelTestCase<TBuffer> {
+  std::vector<std::future<std::shared_ptr<tensorpipe::transport::Connection>>>
+  accept(
+      std::shared_ptr<tensorpipe::transport::Listener> listener,
+      size_t numConnections) {
+    auto promises = std::make_shared<std::vector<
+        std::promise<std::shared_ptr<tensorpipe::transport::Connection>>>>(
+        numConnections);
+    for (size_t i = 0; i < numConnections; ++i) {
+      listener->accept(
+          [&, promises](
+              const tensorpipe::Error& error,
+              std::shared_ptr<tensorpipe::transport::Connection> connection) {
+            ASSERT_FALSE(error) << error.what();
+            connection->read([promises, connection](
+                                 const tensorpipe::Error& error,
+                                 const void* connIdBuf,
+                                 size_t length) {
+              ASSERT_FALSE(error) << error.what();
+              ASSERT_EQ(sizeof(uint64_t), length);
+              promises->at(*static_cast<const uint64_t*>(connIdBuf))
+                  .set_value(std::move(connection));
```

Comment:
If you're moving then the lambda needs to be mutable.

Reply:
Does it? I'm just moving an argument here, not a captured variable.

Reply:
The arguments of this lambda are `error`, `connIdBuf` and `length`. The `connection` was an argument to the "parent" lambda, but it has since been captured into this child lambda (see line 219).